File: exercises/java/003_timers/src/main/java/com/akka/training/barista/BaristaCommand.java
package com.akka.training.barista;

import com.akka.training.Coffee;
import com.akka.training.coffeeMachine.CoffeeMachineCommand;
import static com.akka.training.barista.BaristaCommand.*;

public sealed interface BaristaCommand
    permits OrderCoffee, WrappedCoffeeMachineCoffeeReady {

  record OrderCoffee(String whom, Coffee coffee) implements BaristaCommand {}

  record WrappedCoffeeMachineCoffeeReady(CoffeeMachineCommand.CoffeeReady coffeeReady)
      implements BaristaCommand {}
}

File: exercises/scala/003_timers/src/main/scala/dev/fResult/akka/barista/BaristaCommand.scala
package dev.fResult.akka.barista

import dev.fResult.akka.Coffee

enum BaristaCommand {
  case OrderCoffee(whom: String, coffee: Coffee)
  case CoffeeReady(coffee: Coffee)
  case WrappedCoffeeMachineCoffeeReady(coffeeReady: CoffeeReady)
}
